src/match/views.py

```diff
@@ -0,0 +1,81 @@
+import json
+
+from api import settings as api_settings
+from api.utils import failure_response
+from api.utils import success_response
+from django.db.models import Q
+from match.models import Match
+from match.serializers import BothUsersMatchSerializer
+from match.serializers import MatchSerializer
+from rest_framework import generics
+
+from .controllers.create_match_controller import CreateMatchController
+from .controllers.update_match_controller import UpdateMatchController
+
+
+class MatchesView(generics.GenericAPIView):
+    serializer_class = BothUsersMatchSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def get(self, request):
+        """Get all matches."""
+        matches = Match.objects.all()
+        return success_response(self.serializer_class(matches, many=True).data)
+
+    def post(self, request):
+        """Create a match."""
+        try:
+            data = json.loads(request.body)
```

Comment:
this is what I was referring to about the json loading for the request being redundant in `CreateMatchController` meaning that it's pointless to pass `request` into the controller